src/main/java/bio/terra/service/filedata/FileService.java

```diff
@@ -186,6 +186,29 @@ public FileModel lookupPath(String datasetId, String path, int depth) {
     return fileModelFromFSItem(fsItem);
   }
 
+  public boolean pathExists(String datasetId, String path) {
```

Comment:
Right, I'd do away with this method altogether and just do a lookup

Reply:
I want to keep the interface of the `lookupPath` method the same, because it throws an exception if the file isn't found. I don't want to rely on exceptions for flow-control, so I did this. But, you're right, I could return an `Optional<FileModel>` here and switch based on that being filled rather than a boolean, while returning a possible result at the same time.